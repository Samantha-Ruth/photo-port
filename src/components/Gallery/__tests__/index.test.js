// __tests__/Gallery.test.js
import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Gallery from '..'
const portrait = { name: "portraits", description: "Portraits of people in my life" };


afterEach(cleanup)

describe('Gallery is rendering', () => {


  it('renders', () => {
    render(<Gallery currentCategory={portrait} />);
  });

  it('Matches Snapshot', () => {
    const { asFragment } = render(<Gallery currentCategory={portrait} />)
    expect(asFragment()).toMatchSnapshot()
  })
});

it('h1 tag matches category', () => {
  const { getByTestId } = render(<Gallery currentCategory={portrait} />)
  // eslint-disable-next-line testing-library/prefer-screen-queries
  expect(getByTestId('h1tag')).toHaveTextContent('Portraits')
})