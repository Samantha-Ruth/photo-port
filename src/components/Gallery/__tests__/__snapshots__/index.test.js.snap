// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="h1tag"
    >
      Portraits
    </h1>
    <p>
      Portraits of people in my life
    </p>
    <div>
      <div
        class="flex-row"
      />
    </div>
  </section>
</DocumentFragment>
`;
